WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -472,9 +472,12 @@ else if (mPost.getDate_created_gmt() > 0)
             mPost.setLocation(mPostLocation);
         }
 
+        if (mCategories != null) {
+            mPost.setJSONCategories(new JSONArray(mCategories));
+        }
```

Comment:
> If it's just in a bad state I don't think we'd want to delete the existing categories.

Fair enough. But what about adding null checks in the other places where `mCategories` is accessed?